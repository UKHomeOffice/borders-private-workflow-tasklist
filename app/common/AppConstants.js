
export default class AppConstants {
    static DASHBOARD_PATH = "/dashboard";
    static YOUR_TASKS_PATH = "/your-tasks";
    static YOUR_GROUP_TASKS_PATH = '/your-group-tasks';
    static PROCEDURES_PATH = '/procedures';
    static REPORTS_PATH = '/reports';
    static REPORT_PATH = "/report";
    static MESSAGES_PATH = "/messages";
    static CALENDAR_PATH = "/calendar";
    static START_A_PROCEDURE = "/start-a-procedure";
    static TASK_PATH="/task";
    static ADMIN_PATH="/admin";
    static SHIFT_PATH = "/shift";
    static PROCEDURE_DIAGRAM_PATH = "/procedure-diagram";
    static ONBOARD_USER_PATH="/onboard-user";
    static MANDATORY_DECLARATION_PATH="/mandatory-declarations";
    static  ONE_MINUTE = 60000;

}
