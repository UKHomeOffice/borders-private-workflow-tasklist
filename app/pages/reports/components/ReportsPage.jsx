import React  from 'react';
import PropTypes from 'prop-types';
import {withRouter} from "react-router-dom";
import * as actions from "../actions";
import {createStructuredSelector} from "reselect";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import {loadingReports, reports} from "../selectors";
import DataSpinner from '../../../core/components/DataSpinner';

const uuidv4 = require('uuid/v4');


export class ReportsPage extends React.Component {

    componentDidMount() {
        this.props.fetchReportsList();
    }


    render() {
        const {loadingReports, reports} = this.props;
        const pointerStyle = {cursor: 'pointer'};
        const items = [];
        if (reports) {
            reports.forEach((report) => {
                items.push(<tr key={uuidv4()} id="report" style={pointerStyle} onClick={() => {
                    this.props.history.push(`/report?reportName=${report.get('htmlName')}`)
                }} className="govuk-table__row">
                    <td className="govuk-table__header">{report.get('name')}</td>
                    <td className="govuk-table__cell">{report.get('description')}</td>
                </tr>)
            });
        }

        return <div>
            <div className="govuk-grid-row">
                <div className="govuk-grid-column-one-half">
                    <span className="govuk-caption-l">Operational reports</span>
                    <h2 className="govuk-heading-l">{reports.size} reports</h2>
                </div>

            </div>
            {loadingReports ? <div style={{display: 'flex', justifyContent: 'center', paddingTop: '20px'}}><DataSpinner
                message="Loading reports"/></div> : <div>
                <table className="govuk-table">
                    <tbody className="govuk-table__body">
                    {items}
                    </tbody>
                </table>
            </div>}
        </div>
    }
}


ReportsPage.propTypes = {
    fetchReportsList: PropTypes.func.isRequired,
    loadingReports: PropTypes.bool
};


const mapStateToProps = createStructuredSelector({
    reports: reports,
    loadingReports: loadingReports,

});


const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ReportsPage));
