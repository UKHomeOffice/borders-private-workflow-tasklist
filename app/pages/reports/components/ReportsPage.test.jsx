import React from 'react';
import configureStore from 'redux-mock-store';
import Immutable, { Map, List } from 'immutable';
import Spinner from 'react-spinkit';
import AppConstants from '../../../common/AppConstants';
import { ReportsPage } from './ReportsPage';

describe('Reports Page', () => {
  const initialState = {
    'reports-page': new Map({
      loadingReports: false,
      reports: List([]),
    }),
  };
  const store = configureStore()(initialState);

  it('sets document title as expected', () => {
    const props = {
      loadingReports: false,
      reports: List([]),
      fetchReportsList: jest.fn(),
      store,
    };
    shallow(<ReportsPage {...props} />);
    expect(global.window.document.title).toBe(
      `Operational reports | ${AppConstants.APP_NAME}`,
    );
  });

  it('renders data spinner', async () => {
    const props = {
      loadingReports: true,
      reports: List([]),
      fetchReportsList: jest.fn(),
      store,
    };
    const wrapper = await shallow(
      <ReportsPage {...props} />,
    );
    expect(props.fetchReportsList).toBeCalled();
    expect(wrapper.find('#reportsCountLabel').text()).toEqual('Operational reports0 reports');
    expect(wrapper.containsMatchingElement(Spinner)).toEqual(true);
  });

  it('renders list of reports', async () => {
    const props = {
      loadingReports: false,
      reports: Immutable.fromJS([
        {
          name: 'reportname',
          description: 'reportdescription',
          htmlName: 'reportHtmlName',
        },
      ]),
      fetchReportsList: jest.fn(),
      store,
    };
    const wrapper = await mount(
      <ReportsPage {...props} />,
    );
    expect(props.fetchReportsList).toBeCalled();
    expect(wrapper.find('#reportsCountLabel').text()).toEqual('Operational reports1 report');

    expect(wrapper.find('#report').length).toEqual(1)
  });
});
