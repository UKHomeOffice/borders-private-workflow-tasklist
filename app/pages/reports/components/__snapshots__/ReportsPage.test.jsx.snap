// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports Page matches snapshot 1`] = `
<div>
  <div
    className="govuk-grid-row"
    id="reportsCountLabel"
  >
    <div
      className="govuk-grid-column-one-half"
    >
      <span
        className="govuk-caption-l"
      >
        Operational reports
      </span>
      <h2
        className="govuk-heading-l"
      >
        0
         
        reports
      </h2>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "paddingTop": "20px",
      }
    }
  >
    <DataSpinner
      message="Loading reports"
    />
  </div>
</div>
`;
